### PR TITLE
adding location hierarchy list in client-setting api

### PR DESCRIPTION
**Pull request by neosoftSatish**
adding location hierarchy list in client-setting api

### CODE REVIEW

**Comment by ase-101** [comment]
@neosoftSatish  seems like your commit is included old files as completely new commits. Only your files and your changes should be part of the PR

**Comment by neosoftSatish** [comment]
> @neosoftSatish seems like your commit is included old files as completely new commits. Only your files and your changes should be part of the PR

I have added only those files which were not present in the project. I have not committed any old files.

File: admin/kernel-masterdata-service/src/main/java/io/mosip/kernel/masterdata/controller/LocationHierarchyController.java [hunk 1]
```diff
@@ -57,13 +64,36 @@
...
 	}
 
+	/**
+	 * 
+	 * Function to fetch location Hierarchy Levels details
+	 * 
+	 * @return LocationHierarchyLevelResponseDto Location Hierarchy Levels from DB
+	 *         {@link LocationHierarchyLevelResponseDto}
+	 */
+	@PreAuthorize("hasAnyRole('ZONAL_ADMIN','GLOBAL_ADMIN','INDIVIDUAL','PRE_REGISTRATION','REGISTRATION_PROCESSOR')")
```
Comment (ase-101) [inline admin/kernel-masterdata-service/src/main/java/io/mosip/kernel/masterdata/controller/LocationHierarchyController.java:74]: Add Default role

File: admin/kernel-syncdata-service/src/main/java/io/mosip/kernel/syncdata/config/Config.java [hunk 2]
```diff
@@ -48,6 +50,12 @@
 	public Filter registerCORSFilter() {
 		return new CorsFilter();
 	}
+	
+	@Primary
```
Comment (ase-101) [inline admin/kernel-syncdata-service/src/main/java/io/mosip/kernel/syncdata/config/Config.java:54]: remove this code, and 
private RestTempalate restTemplate

File: admin/kernel-syncdata-service/src/main/java/io/mosip/kernel/syncdata/config/MosipEnvironment.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.EnvironmentAware;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.env.Environment;
+
+@Configuration
+public class MosipEnvironment implements EnvironmentAware {
+
+	@Autowired
+	private Environment environment;
```
Comment (ase-101) [inline admin/kernel-syncdata-service/src/main/java/io/mosip/kernel/syncdata/config/MosipEnvironment.java:12]: directly inject required property with @Value("${}") in the required class, Please remove this class

File: admin/kernel-syncdata-service/src/main/java/io/mosip/kernel/syncdata/utils/SyncMasterDataServiceHelper.java [hunk 4]
```diff
@@ -279,6 +290,16 @@
 	@Value("${mosip.syncdata.tpm.required:false}")
 	private boolean isTPMRequired;
 
+	@Autowired
+	MosipEnvironment mosipEnvironment;
+
+	@Qualifier("authRestTemplate")
```
Comment (ase-101) [inline admin/kernel-syncdata-service/src/main/java/io/mosip/kernel/syncdata/utils/SyncMasterDataServiceHelper.java:296]: @qualifier not required, restTemplate bean is already available

File: admin/kernel-syncdata-service/src/main/java/io/mosip/kernel/syncdata/utils/SyncMasterDataServiceHelper.java [hunk 5]
```diff
@@ -330,6 +351,63 @@
...
+		}
+
+		ResponseEntity<String> response = restTemplate.exchange(builder.toUriString(), HttpMethod.GET,
+				new HttpEntity<Object>(headers), String.class);
+		if (response.getStatusCode().equals(HttpStatus.OK)) {
+			String responseBody = response.getBody();
+			List<ServiceError> validationErrorsList = null;
+			validationErrorsList = ExceptionUtils.getServiceErrorList(responseBody);
+			if (!validationErrorsList.isEmpty()) {
+				// throw new SyncServiceException(validationErrorsList);
```
Comment (ase-101) [inline admin/kernel-syncdata-service/src/main/java/io/mosip/kernel/syncdata/utils/SyncMasterDataServiceHelper.java:390]: you need to throw this exception

File: admin/kernel-syncdata-service/src/main/resources/application-local.properties [hunk 6]
```diff
@@ -16,6 +16,8 @@ mosip.kernel.keymanager-service-publickey-url=https://dev.mosip.net/v1/keymanage
 mosip.kernel.syncdata.cryptomanager-encrypt-url=https://dev.mosip.net/v1/cryptomanager/private/encrypt
 mosip.kernel.syncdata.public-key-url=https://dev.mosip.net/v1/keymanager/publickey/{applicationId}
 
+mosip.kernel.idobjectvalidator.masterdata.locationhierarchylevels.rest.uri = https://dev.mosip.net/v1/masterdata/locationHierarchyLevels
```
Comment (ase-101) [inline admin/kernel-syncdata-service/src/main/resources/application-local.properties:19]: mosip.kernel.masterdata.locationhierarchylevels.url=https://dev.mosip.net/v1/masterdata/locationHierarchyLevels

**Review by ase-101 — CHANGES_REQUESTED** [review]

**Pull request merged**